src/com/lftechnology/java/training/alina/collections/generics/instantiateproduct/GenericClass.java

```diff
@@ -0,0 +1,32 @@
+
+package com.lftechnology.java.training.alina.collections.generics.instantiateproduct;
+
+/**
+ * GenericClass consists of Class definition of classes
+ * 
+ * @author Alina Shakya <alinashakya@lftechnology.com>
+ * @param <T>
+ */
+public class GenericClass<T> {
+
+	private Class<T> classDetail;
+
+	public GenericClass(Class<T> classDetail) {
+
+		this.classDetail = classDetail;
+	}
+
+	/**
+	 * Builds new instance class
+	 * 
+	 * @return
+	 * @throws InstantiationException
+	 * @throws IllegalAccessException
+	 * @author Alina Shakya <alinashakya@lftechnology.com>
+	 */
+	public T buildOne()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.lang.InstantiationException, java.lang.IllegalAccessException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1160)